code/obj/item/gun/kinetic.dm

```diff
@@ -2144,6 +2144,209 @@ ABSTRACT_TYPE(/obj/item/survival_rifle_barrel)
 		else
 			boutput(user, SPAN_ALERT("You can't fire \the [src] when it is open!"))
 
+#define ONE_BARREL 1
+#define TWO_BARRELS 2
+#define ALL_BARRELS 4
+#define MAX_USES 9
+
+/obj/item/gun/kinetic/sawnoff/quadbarrel //for salvagers
+
+	name = "\improper Four Letter Word"
+	desc = "For when you REALLY need to get the point across."
+	icon = 'icons/obj/large/64x32.dmi'
+	icon_state = "coachgun" //for sake of not rewriting a ton of break action code, it uses the same names as the DB, but in a different DMI. (change?)
+	item_state = "quadbarrel" //custom inhands, though.
+
+	camera_recoil_sway_min = 10 //VIOLENCE!
+	recoil_strength = 15
+	recoil_max = 60
+
+
+	var/guaranteed_uses = MAX_USES	// allows for just over two volleys before it starts breaking
+	var/firemode = ONE_BARREL
+
+	force = MELEE_DMG_RIFLE
+	max_ammo_capacity = 4
+	two_handed = TRUE
+
+	New()
+		..()
+		ammo = new/obj/item/ammo/bullets/pipeshot/plasglass
+		set_current_projectile(new/datum/projectile/special/spreader/buckshot_burst/plasglass)
+		name = "Four Letter Word" //I kinda like the fact that it'll pull from the DB name pool buuut I kinda don't.
+		UpdateIcon()
+
+	get_help_message(dist, mob/user)
+		.+= "You can use a <b>welding tool</b> to repair its state. \n You can also use a <b>screwdriver</b> to cycle firing modes."
+
+	examine()
+		. = ..()
+		. += "\n \n It is set to fire " //differentiate the FLW examine text from the default gun examine text
+		switch(firemode)
+			if(ONE_BARREL)
+				. += "one barrel."
+			if(TWO_BARRELS)
+				. += "two barrels."
+			if(ALL_BARRELS)
+				. += "all four barrels!"
+		if (guaranteed_uses == MAX_USES)
+			. += "It's in perfect condition!"
+		else if (guaranteed_uses <= 0) //negative damage
+			. += " It seems severely damaged!"
+		else if (guaranteed_uses < 4) //0-4
+			. += " It seems pretty damaged."
+		else if(guaranteed_uses < 7) //4-7
+			. += " It's damaged."
+		else  //7+
+			. += " It's barely damaged."
+
+	shoot(var/target, var/start, var/mob/user)
+
+		var/i //Because we can't initialize variables when we're declaring the loop. thanks, DM.
+		var/priorammo = src.ammo.amount_left
+
+		//Go up to the limit defined by the firing mode, but don't exceed however many bullets are in the gun BEFORE we started firing
+		for(i=1, ((i <= priorammo) && (i <= firemode)), i++)
+			..() //shoot an additional ith time
+			guaranteed_uses-- //damage the gun an additional ith time
+
+		//you're shooting multiple shotgun shells out of a garbage gun at the same time. don't think there won't be consequences
+		if((firemode == TWO_BARRELS) && (priorammo != 0))
```

Comment:
Should this occur iff there are 2 rounds shot?

Reply:
It can be removed, but a well placed 2-round burst can do a flat 100 if every shot hits. Ideally, I want to discourage its use in melee for that reason.